application/image_processor.py

```diff
@@ -1,46 +1,223 @@
+import datetime
+import os
+
 import requests
 import torch
+import validators
 from PIL import Image
 from transformers import BlipProcessor, BlipForConditionalGeneration
 
+import utils
+
+
+def caption_image(img: Image, img_source: str, processor, model) -> None:
+    """
+    This method will caption the given Pillow Image using the given model, which is the Salesforce captioning model.
+    Two captions will be generated and outputted.
+    :param img:
+    :param img_source:
+    :param processor:
+    :param model:
+    :return: None
+    """
+    # Conditional image captioning
+    text = "a photo of"
+    inputs = processor(img, text, return_tensors="pt")
+
+    out = model.generate(**inputs, max_length=60)
+    con_caption: str = 'Conditional image caption: ' + processor.decode(out[0], skip_special_tokens=True)
+    print(con_caption)
+
+    # Unconditional image captioning
+    inputs = processor(images=img, return_tensors="pt")
+
+    out = model.generate(**inputs, max_length=60)
+    unc_caption: str = 'Unconditional image caption: ' + processor.decode(out[0], skip_special_tokens=True)
+    print(unc_caption, end='\n\n')
+
+    store_image(create_dir(), img_source, con_caption, unc_caption)
+
+
+def store_image(path: str, img_source: str, con_caption: str, unc_caption: str) -> None:
+    """
+    Stores the given captions of the given image in a file. Creates a new folder on the local device if that path
+    doesn't already exist.
+    :param path: The path of the directory
+    :param img_source: The URL or directory path that sourced the image
+    :param con_caption: Conditional caption
+    :param unc_caption: Unconditional caption
+    :return: None
+    """
+
+    now: datetime = datetime.datetime.now()
```

Comment:
Personal preference pt2: rename to current_time

Reply:
Made name change.